cypress/support/API/leaveAPI/leaveAPI.ts

```diff
@@ -0,0 +1,45 @@
+export default class leave {
+  static adminAddLeaveEntitlements(empNum: number) {
+    cy.request({
+      method: "POST",
+      url: "/api/v2/leave/leave-entitlements",
+      body: {
+        empNumber: empNum,
+        leaveTypeId: 7,
+        fromDate: "2023-01-01",
+        toDate: "2023-12-31",
+        entitlement: "100",
+      },
+    }).then((res) => {
+      cy.logout();
+    });
+  }
+
+  static userAddRequestLeave() {
+    return cy.request({
+      method: "POST",
+      url: "/api/v2/leave/leave-requests",
+      body: {
+        leaveTypeId: 7,
+        fromDate: "2023-12-05",
+        toDate: "2023-12-06",
+        comment: null,
```

Comment:
why you pass the body as constant value?! you should pass them as variable 
And for the dates you should use the moment js library we used it to declare the dates,
for these dates for the future case, when you write the test in that day that was worked but when run test after (6-12-2023) Not worked cuz 6 was past dates so not everyday the test will working , so please use moment